fe1-web/src/features/witness/components/WitnessNotification.tsx

```diff
@@ -67,17 +103,80 @@ const WitnessNotification = ({ notification, navigateToNotificationScreen }: IPr
       navigateToNotificationScreen();
     }
   };
-
   return (
-    <View>
-      <Text>{JSON.stringify(notification)}</Text>
-      <Text>{JSON.stringify(message)}</Text>
-      <PoPTextButton onPress={onWitness} disabled={!isConnected}>
-        {STRINGS.witness_message_witness}
-      </PoPTextButton>
-      <PoPTextButton onPress={onDecline} disabled={!isConnected}>
-        {STRINGS.meeting_message_decline}
-      </PoPTextButton>
+    <View style={styles.container}>
+      <Text style={[Typography.base, Typography.important, styles.marginB10]}>
+        {STRINGS.witness_req}
+      </Text>
+      {decodedData ? (
+        <>
+          {decodedData.object === 'roll_call' && decodedData.action === 'create' ? (
+            <>
+              <View style={styles.marginB15}>
+                <Text style={[Typography.small, styles.boldText]}>
+                  {decodedData.object}#{decodedData.action}:
+                </Text>
+                <Text style={Typography.small}>Name: {decodedData.name}</Text>
+                <Text style={Typography.small}>ID: {decodedData.id}</Text>
+                <Text style={Typography.small}>
+                  Created at: {new Date(decodedData.creation * 1000).toLocaleString()}
+                </Text>
+                <Text style={Typography.small}>
+                  Proposed start: {new Date(decodedData.proposed_start * 1000).toLocaleString()}
+                </Text>
+                <Text style={Typography.small}>
+                  Proposed end: {new Date(decodedData.proposed_end * 1000).toLocaleString()}
+                </Text>
+                <Text style={Typography.small}>Location: {decodedData.location}</Text>
+                {decodedData.description ? (
+                  <Text style={Typography.small}>Description: {decodedData.description}</Text>
+                ) : (
+                  <Text />
+                )}
+              </View>
+              <View style={styles.marginB15}>
+                <Text style={[Typography.small, styles.boldText]}>Message Information:</Text>
+                <Text style={Typography.small}>Message ID: {notification.messageId}</Text>
+                <Text style={Typography.small}>Received from: {message.receivedFrom}</Text>
+                <Text style={Typography.small}>Channel: {message.channel}</Text>
+                <Text style={Typography.small}>Sender: {message.sender}</Text>
+                <Text style={Typography.small}>Signature: {message.signature}</Text>
+                <Text style={Typography.small}>
```

Comment:
If you use the same style everywhere I think you can wrap the <Text> elements inside another <Text> elements like here: https://reactnative.dev/docs/text#nested-text. Saves you same duplication :)